src/components/graph/graph.ts

```diff
@@ -519,56 +522,53 @@ export default class GraphComponent {
     const transformRaw = this._g.attr('transform')
-    if (transformRaw === null) return {translation: {x: 0, y: 0}, scale: 1}
+    const transform: ITransform = {translation: {x: 0, y: 0}, scale: 1}
+    if (transformRaw === null) return {transform}
     const [translationRaw, scaleRaw] = transformRaw.split(' ')
-    let translation = translationRaw
+    const translationValues = translationRaw
       .replace('translate(', '')
       .replace(')', '')
       .split(',')
       .map(Number)
-    translation = {x: translation[0], y: translation[1]}
-    const scale = Number(scaleRaw.match(/\d(\.\d+)*/)[0])
-    return {translation, scale}
+    transform.translation = {x: translationValues[0], y: translationValues[1]}
+    transform.scale = Number(scaleRaw.match(/\d(\.\d+)*/)[0])
+    return {transform}
   }
 
-  private _graphToSVGPosition(d: {x: number, y: number})
-    : {position: {x: number, y: number}} {
+  private _graphToSVGPosition(d: {x: number, y: number}): {x: number, y: number} {
     const {corners} = this._getGraphCornerPoints()
     const position = {
       x: this._width * (d.x - corners.minX) / (corners.maxX - corners.minX),
       y: this._height * (d.y - corners.minY) / (corners.maxY - corners.minY),
     }
-    return {position}
+    return position
   }
 
-  private _svgToGraphPosition(d: {x: number, y: number})
-    : {position: {x: number, y: number}} {
+  private _svgToGraphPosition(d: {x: number, y: number}): {x: number, y: number} {
     const {corners} = this._getGraphCornerPoints()
     const position = {
       x: (d.x / this._width) * (corners.maxX - corners.minX) + corners.minX,
       y: (d.y / this._height) * (corners.maxY - corners.minY) + corners.minY,
     }
-    return {position}
+    return position
   }
 
-  private _getGraphCornerPoints()
-    : {corners: {minX: number, maxX: number, minY: number, maxY: number}} {
-    const {translation, scale} = this._getGraphTranslationAndScale()
+  private _getGraphCornerPoints(): {corners: ICorner} {
+    const {transform} = this._getGraphTranslationAndScale()
     const corners = {
-      minX: -translation.x / scale,
-      maxX: this._width / scale - translation.x / scale,
-      minY: -translation.y / scale,
-      maxY: this._height / scale - translation.y / scale,
+      minX: -transform.translation.x / transform.scale,
+      maxX: this._width / transform.scale - transform.translation.x / transform.scale,
+      minY: -transform.translation.y / transform.scale,
+      maxY: this._height / transform.scale - transform.translation.y / transform.scale,
     }
     return {corners}
   }
 
   private _graphTransformToString(): {str: string} {
-    const {translation, scale} = this._getGraphTranslationAndScale()
-    const str = translation.x + ' ' + translation.y + ' ' + scale
+    const {transform} = this._getGraphTranslationAndScale()
+    const str ='${transform.translation.x} ${transform.translation.y} ${transform.scale}'
```

Comment:
backticks, not single quotes for template strings. This form gives you a literal '${transform.translation.y}' in your resultant string.